### PR TITLE
Leave room on Disconnect + Close Media Room

### PR DESCRIPTION
**Pull request by xmquentin**
Lol so many bugs, which I think are all RFE related. Those will be talked about in our next meeting, but for right now, leaving the media room on disconnect and then re-joining works for the following:

* If userN leaves the room and re-joins for N people in the room originally

This means if user2 leaves while user1 and user3 are in the room and re-joins, things break on the RFE side. I could not quickly solve those.

This is at least some progress!

### CODE REVIEW

**Comment by xmquentin** [comment]
Decided to add CloseRoom functionality in here

File: server/server.go [hunk 1]
```diff
@@ -195,7 +242,7 @@ func (r *RoomzSignalingServer) leaveMediaRoomHandler(s socketio.Conn, data map[s
     }
   }
   if delIdx >= 0 {
-    log.Printf("Removed peerId=%s from roomId=%d", peerId, roomId)
+    log.Printf("%s Removed peerId=%s from roomId=%d", prefix, peerId, roomId)
     r.roomUsers[roomId] = append(r.roomUsers[roomId][:delIdx], r.roomUsers[roomId][delIdx+1:]...)
   }
   r.roomUsersMtx.Unlock()
```
Comment (hridayeshvjoshi) [inline server/server.go:248]: LGTM

**Review by hridayeshvjoshi — COMMENTED** [review]
LGTM

**Review by ABFranco — APPROVED** [review]
lgtm

**Pull request merged**